src/eth.rs

```diff
@@ -122,6 +122,19 @@ pub struct EIP1559Transaction {
     pub data: Vec<u8>,
 }
 
+/// Identifies the case of the recipient address given as hexadecimal string.
+/// This function exists as a convenience to potentially help clients to determine the case of the
+/// recipient address.
+pub fn eth_identify_case(recipient_address: &str) -> pb::EthAddressCase {
```

Comment:
ChatGPT suggests this improvement to not allocate a String:

```
pub fn eth_identify_case(recipient_address: &str) -> pb::EthAddressCase {
    if recipient_address.chars().all(|c| c.is_ascii_uppercase()) {
        pb::EthAddressCase::Upper
    } else if recipient_address.chars().all(|c| c.is_ascii_lowercase()) {
        pb::EthAddressCase::Lower
    } else {
        pb::EthAddressCase::Mixed
    }
}
```

Please also add a `test_eth_identify_case()` unit test in the `mod tests {` block at the bottom of the file. ChatGPT can probably make this quickly too :smile:

Reply:
fixed, but had to add `!c.is_ascii_alphabetic()` since the original one doesn't work with numbers

```
    if recipient_address
        .chars()
        .all(|c| !c.is_ascii_alphabetic() || c.is_ascii_uppercase())
    {
        pb::EthAddressCase::Upper
    } else if recipient_address
        .chars()
        .all(|c| !c.is_ascii_alphabetic() || c.is_ascii_lowercase())
    {
        pb::EthAddressCase::Lower
    } else {
        pb::EthAddressCase::Mixed
    }
```